lib/elixir/src/elixir_expand.erl

```diff
@@ -891,6 +892,29 @@ assert_generator_start(Meta, _, E) ->
 
 %% Assertions
 
+refute_parallel_bitstring_match({'<<>>', _, _}, {'<<>>', Meta, _} = Arg, E, true) ->
+  form_error(Meta, ?key(E, file), ?MODULE, {parallel_bitstring_match, Arg});
+refute_parallel_bitstring_match(Expr, {'=', _Meta, [Left, Right]}, E, Parallel) ->
+  refute_parallel_bitstring_match(Expr, Left, E, true),
+  refute_parallel_bitstring_match(Expr, Right, E, Parallel);
+refute_parallel_bitstring_match(Expr, Arg, E, Parallel) when is_list(Expr), is_list(Arg) ->
+  refute_parallel_bitstring_match_list(Expr, Arg, E, Parallel);
+refute_parallel_bitstring_match(Expr, Arg, E, Parallel) when tuple_size(Expr) == 2, tuple_size(Arg) == 2 ->
+  refute_parallel_bitstring_match_list(tuple_to_list(Expr), tuple_to_list(Arg), E, Parallel);
+refute_parallel_bitstring_match({'{}', _, Args1}, {'{}', _, Args2}, E, Parallel) ->
+  refute_parallel_bitstring_match_list(Args1, Args2, E, Parallel);
+refute_parallel_bitstring_match(_Expr, _Arg, _E, _Parallel) ->
+  ok.
+
+refute_parallel_bitstring_match_list([], _Arg, _E, _Parallel) ->
+  ok;
+refute_parallel_bitstring_match_list(_Expr, [], _E, _Parallel) ->
+  ok;
+refute_parallel_bitstring_match_list([Arg1 | Rest1], [Arg2 | Rest2], E, Parallel) ->
+  refute_parallel_bitstring_match(Arg1, Arg2, E, Parallel),
+  refute_parallel_bitstring_match_list(Rest1, Rest2, E, Parallel).
```

Comment:
Does all of this belong here, or should we move it to `elixir_bitstring`? Just wondering.

Reply:
@whatyouhide considering that `=` is what triggers the error I think it belongs here.